auth/auth.go

```diff
@@ -723,6 +737,24 @@ func (auth *Authenticator) authenticateOIDCIdentity(identity *Identity, provider
 	}
 	base.DebugfCtx(auth.LogCtx, base.KeyAuth, "OIDCUsername: %v", base.UD(username))
 
+	var oidcRoles, oidcChannels base.Set
+	if provider.RolesClaim != "" {
+		oidcRoles, err = getJWTClaimAsSet(identity, provider.RolesClaim)
+		if err != nil {
+			return nil, time.Time{}, fmt.Errorf("failed to find OIDC roles: %w", err)
```

Comment:
getJWTClaimAsSet only returns an error if the claim is present, but has a type other than `string` or `[]string` (if it's simply absent it returns the empty set), which to me would indicate a misconfiguration on the provider side - in my head it'd be confusing why authentication succeeds but the user doesn't get the channels they're supposed to, so having it fail would highlight it faster (though I could definitely see the case for succeeding but logging the error).

Reply:
I've thought about it further and think I agree (about fail fast).  In the scenario where claims are incorrectly defined on the provider side, it would be preferable for auth to fail, as opposed to replications moving forward with gaps in data because of (possibly temporary) issues with claim-based channel assignment.